### DISCUSSION TITLE
16KB granules

### DISCUSSION THREAD

**Discussion by evansm7** [evansm7/vftool#9] [Show and tell]
The M1 hardware supports 16KB translation granules (AKA base page size) as well as the usual 4KB.  A larger page makes more efficient use of the TLB and can lead to higher performance. 

I use a mainline Linux 5.9 kernel with my Debian VM, and rebuilding it to use 16KB granules gave me a 16% performance improvement when compiling a kernel inside the VM. (And the VM is very fast already!)

Has anyone else tried this, and if so what effects did you see? What other workloads benefit?

**Comment by caldrian** [comment]
Just quickly tried it out, works like a charm! Thanks for the hint.
